client/task-list/tasks/payments/RemotePayments/components/PaymentMethod.js

```diff
@@ -30,63 +30,59 @@ export const PaymentMethod = ( {
 	const { key, plugins, title } = method;
 	const slot = useSlot( `woocommerce_remote_payment_${ key }` );
 	const hasFills = Boolean( slot?.fills?.length );
-	const [ isFetchingPaymentGateway, setIsFetchingPaymentGateway ] = useState(
-		false
-	);
-	const [ paymentGateway, setPaymentGateway ] = useState( null );
+	const [ isPluginLoaded, setIsPluginLoaded ] = useState( false );
 
 	useEffect( () => {
 		recordEvent( 'payments_task_stepper_view', {
 			payment_method: key,
 		} );
 	}, [] );
 
-	const { activePlugins } = useSelect( ( select ) => {
-		const { getActivePlugins } = select( PLUGINS_STORE_NAME );
+	const {
+		isOptionUpdating,
+		isPaymentGatewayResolving,
+		needsPluginInstall,
+		paymentGateway,
+	} = useSelect( ( select ) => {
+		const { isOptionsUpdating } = select( OPTIONS_STORE_NAME );
+		const { getPaymentGateway, isResolving } = select(
+			PAYMENT_GATEWAYS_STORE_NAME
+		);
+		const activePlugins = select( PLUGINS_STORE_NAME ).getActivePlugins();
+		const pluginsToInstall = plugins.filter(
+			( m ) => ! activePlugins.includes( m )
+		);
 
 		return {
-			activePlugins: getActivePlugins(),
+			isOptionUpdating: isOptionsUpdating(),
+			isPaymentGatewayResolving: isResolving( 'getPaymentGateway', [
+				key,
+			] ),
+			paymentGateway: ! pluginsToInstall.length
+				? getPaymentGateway( key )
+				: null,
+			needsPluginInstall: !! pluginsToInstall.length,
 		};
 	} );
 
-	const isOptionsRequesting = useSelect( ( select ) => {
-		const { isOptionsUpdating } = select( OPTIONS_STORE_NAME );
-
-		return isOptionsUpdating();
-	} );
-
-	const pluginsToInstall = plugins.filter(
-		( m ) => ! activePlugins.includes( m )
-	);
-
 	useEffect( () => {
-		if (
-			pluginsToInstall.length ||
-			paymentGateway ||
-			isFetchingPaymentGateway
-		) {
+		if ( ! paymentGateway ) {
 			return;
 		}
-		fetchGateway();
-	}, [ pluginsToInstall ] );
 
-	// @todo This should updated to use the data store in https://github.com/woocommerce/woocommerce-admin/pull/6918
-	const fetchGateway = () => {
-		setIsFetchingPaymentGateway( true );
-		apiFetch( {
-			path: 'wc/v3/payment_gateways/' + key,
-		} ).then( async ( results ) => {
-			const { post_install_scripts: postInstallScripts } = results;
-			if ( postInstallScripts ) {
-				const scriptPromises = postInstallScripts.map( ( script ) =>
-					enqueueScript( script )
-				);
-				await Promise.all( scriptPromises );
-			}
-			setPaymentGateway( results );
-			setIsFetchingPaymentGateway( false );
-		} );
-	};
+		const { post_install_scripts: postInstallScripts } = paymentGateway;
+		if ( postInstallScripts && postInstallScripts.length ) {
+			const scriptPromises = postInstallScripts.map( ( script ) =>
+				enqueueScript( script )
+			);
+			Promise.all( scriptPromises ).then( () => {
```

Comment:
It looks like this promise is used to keep the Stepper in a "pending" state, but I wonder if it should also be a dependency on the install step? I haven't looked at exactly how the pending state works, but if it proceeds to the next step before a previously unloaded Slot fill takes effect that could be a bit jarring.

Perhaps that's already handled by the pending state, however?

It's also likely to always be a bit jarring, so pretty minor thought here.

Reply:
This does take into account the install step.  Above this:

```js
		if ( ! paymentGateway ) {
			return;
		}
```

So if a gateway hasn't been installed, activated, and fetched we won't begin the script enqueue.  Because the install step could be directly bypassed, this needs to occur at the beginning of the config step.

Reply:
>It's also likely to always be a bit jarring, so pretty minor thought here.

I agree on the current loading states being jarring.  Not entirely sure what we can do about this, but hoping to tackle some of it this sprint.